src/main/java/org/openrewrite/java/migrate/javax/AddTransientAnnotationToPrivateAccessor.java

```diff
@@ -0,0 +1,126 @@
+                            maybeAddImport("javax.persistence.Transient");
+                            return JavaTemplate.builder("@Transient")
+                                    .contextSensitive()
+                                    .javaParser(JavaParser.fromJavaVersion().classpathFromResources(ctx, "javax.persistence-api-2.2"))
+                                    .imports("javax.persistence.Transient")
+                                    .build()
+                                    .apply(getCursor(), md.getCoordinates().addAnnotation(Comparator.comparing(J.Annotation::getSimpleName)));
+                        }
+                        return md;
+                    }
+
+                    private boolean isPrivateAccessorMethodWithoutTransientAnnotation(J.MethodDeclaration method) {
+                        return method.hasModifier(J.Modifier.Type.Private)
+                               && method.getParameters().get(0) instanceof J.Empty
+                               && method.getReturnTypeExpression().getType() != JavaType.Primitive.Void
+                               && FindAnnotations.find(method, "javax.persistence.Transient").isEmpty()
+                               && methodReturnsFieldFromClass(method);
+                    }
+
+                    /**
+                     * Check if the given method returns a field defined in the parent class
+                     */
+                    private boolean methodReturnsFieldFromClass(J.MethodDeclaration method) {
+                        // Get all return statements in method
+                        List<J.Return> returns = new ArrayList<>();
+                        JavaIsoVisitor<List<J.Return>> returnGetter = new JavaIsoVisitor<List<J.Return>>() {
+                            @Override
+                            public J.Return visitReturn(J.Return ret, List<J.Return> statements) {
+                                statements.add(ret);
+                                return super.visitReturn(ret, statements);
+                            }
+                        };
+                        returnGetter.visitBlock(method.getBody(), returns);
+
+                        // Get all return values
+                        List<?> returnValues = returns.stream()
+                                .map(J.Return::getExpression)
+                                .filter(Objects::nonNull)
+                                .map(expression -> {
+                                    if (expression instanceof J.FieldAccess) {
+                                        return ((J.FieldAccess) expression).getName().getFieldType();
+                                    } else if (expression instanceof J.Identifier) { // ie: return field;
+                                        return ((J.Identifier) expression).getFieldType();
+                                    } else {
+                                        return null;
+                                    }
```

Comment:
In a way we're iterating over returns twice; first as we visit each return, and then again to extract the returned field type; would it make sense to fold those into one by doing this extract in the visitor just above?

Reply:
Great catch, will update to do that. Also I noticed that I'm collecting the class variables in each method, so I'll extract that to a higher scope variable.